### ISSUE TITLE
Classes generated by PFL's Wrapper are sometimes incompatible with JDK17

### ISSUE THREAD

**Issue by dmatej** [eclipse-ee4j/glassfish#23628]
When I implemented https://github.com/eclipse-ee4j/glassfish/pull/23621 changes, I tried to follow instructions on @deprecated methods of the Wrapper class, but tests then failed. 
```
// WORKS, but is deprecated:
return Wrapper._generate(loader, generator.getAnchorClass().getProtectionDomain(), props);
//  WORKS, but result is not usable for generating proxy by another classloader
return Wrapper._generate(generator.getAnchorClass(), props);
```

Btw we already have javaassist in dependencies.

This issue relates also to #23627, because on modern JDK's that mess between packages brings serious issues around modularity and classloading rules.

**Comment by dmatej** [comment]
I switched this issue to a bug, because the solution in #23621 doesn't play well with proxies. That was discovered when I resuscitated ejb_all tests under #23507 , so it will go both under single PR, because merged state always have to pass all tests and has to be stable.

The issue is that the generating code by MethodHandles is more strict than the code generated by ClassLoader.defineClass. 
Facts:

* Under JDK11+ doesn't exist Unsafe.defineClass (already resolved in PFL).
* MH: generated class will have the same package and same classloader as the anchorClass
* CL: generated class can have any package, will use the provided classloader and protectiondomain
* getClassLoader methods can return null for bootstrap classloader 
* Proxy.newProxyInstance has strict rules: all referred types must be loadable by the used classloader
* org.glassfish.pfl.dynamic.codegen.spi.Type.type uses internal cache to remember classes while it generates new ones.
* -> no way to generate proxies for class referring class generated by MH for anchorClass loaded by embeddedGF's ASURLCL
* -> EXCEPT ...
